drakrun/drakrun/util.py

```diff
@@ -133,6 +133,37 @@ def safe_delete(file_path) -> bool:
         return False
 
 
+def try_subprocess(list_args: list, msg: str, reraise=True, debug=True, **kwargs) -> subprocess.CompletedProcess:
+
+    # setup default parameters to be passed, can be overrided by supplying kwargs
+    if kwargs.get('stdout') is None:
+        kwargs['stdout'] = subprocess.PIPE
+    elif kwargs.get('stdout') == subprocess.STDOUT:
+        del kwargs['stdout']
```

Comment:
Why do we delete this flag?

Reply:
Ah, so basically we get a `OSError: [Errno 9] Bad file descriptor` if we assign `stdout=subprocess.STDOUT` during subprocess calls, it is not to be assigned actually, but if we want to imply it somehow during multiple `kwargs` calls ( if the default is being used as `subprocess.PIPE` but we want to show the output on the terminal ), I thought that we can give this argument which will be handled internally by `try_run`. Though, I should add a comment here ig, it wasn't really intuitive

Reply:
Hmm, it seems that default behavior is to just pass stdout to stdout. That's why setting it explicitly raises an exception. If one wants to handle the output, `capture_output=True` should be passed.
```
In [15]: subprocess.run(["whoami"])
chivay
Out[15]: CompletedProcess(args=['whoami'], returncode=0)

In [16]: subprocess.run(["whoami"], capture_output=True)
Out[16]: CompletedProcess(args=['whoami'], returncode=0, stdout=b'chivay\n', stderr=b'')
```

Reply:
> Hmm, it seems that default behavior is to just pass stdout to stdout. That's why setting it explicitly raises an exception. If one wants to handle the output, `capture_output=True` should be passed.
> 
> ```
> In [15]: subprocess.run(["whoami"])
> chivay
> Out[15]: CompletedProcess(args=['whoami'], returncode=0)
> 
> In [16]: subprocess.run(["whoami"], capture_output=True)
> Out[16]: CompletedProcess(args=['whoami'], returncode=0, stdout=b'chivay\n', stderr=b'')
> ```

hmmm, so then should i remove `stdout` and `stderr` dependencies and just depend on `check_output`? with seperate dependencies, we can capture just `stderr` output whenever required but show `stdout` on terminal, but if we use `check_output` both of them will be bound to either terminal or capture, and also, we won't be able to redirect `stderr` to `subprocess.STDOUT` which is also used sometimes depending on scenarios, what's your take on all this?

Reply:
I think that current implementation is good.

As for ` elif kwargs.get('stdout') == subprocess.STDOUT:` 
I think that we can ignore this case. If it's an error for run(), we shouldn't handle it.

Reply:
> I think that current implementation is good.
> 
> As for ` elif kwargs.get('stdout') == subprocess.STDOUT:`
> We can ignore this case. If it's an error for run(), we shouldn't handle it.

Umm, then how do you propose redirecting to terminal? Right now, the default is `subprocess.PIPE` if no kwargs are passed ( in the `if` condition above the `elif` you are talking about ) since we don't need everything to be shown in terminal.

Reply:
subpocess.STDOUT seems to be an illegal constant for subprocess.run.
I think that redirection to the terminal can be done by passing `stdout=None` (it's the way subprocess.run behaves)

Reply:
> I think that redirection to the terminal can be done by passing `stdout=None` (it's the way subprocess.run behaves)

Ah, didn't know this, will change it, Thanks!